stl/inc/atomic

```diff
@@ -354,8 +363,14 @@ struct _Atomic_storage {
         _Check_memory_order(_Order);
         const auto _Storage_ptr  = _STD addressof(_Storage);
         const auto _Expected_ptr = _STD addressof(_Expected);
+#if _CMPXCHG_MASK_OUT_PADDING_BITS
+        __builtin_zero_non_value_bits(_Expected_ptr);

```

Comment:
I would prefer to see this just call the intrinsic on `store` / `exchange` / etc. and I disagree that it can't be done in the atomic_ref ctor. It does mean the ctor will have a CAS but very few types people are likely to use with atomics will be non-`has_unique_object_representations`, and for such types this builtin has no effect.

Reply:
It can be done with `lock AND` for most types.
Anyway, why do you want it?

I expect that if CAS uses previously loaded value it does not iterate more than once: non-value bytes would atch on the first try.

Also not handling this directly in `atomic_ref` CAS will break non-standard, but still possibly expected and relied upon behavior that the object referenced by `atomic_ref` can be accessed not by `atomic_ref` even during `atomic_ref` lifetine.

Reply:
CAS is generally a rare operation, normally programs get away with `store` and `exchange`, I would not want to penalize other operation for something that is not used mos of the time.
`atomic_ref` constructor is supposed to be no-op, the proposal directly mentions processing non-atomic array for high-performance computing.

Maybe it is better to try reach proposal authors before going further

Reply:
If the concern is keep normal atomics optimized, then they are kept optimized in release mode, compiler skips both the loop, and xor comparison with mask.

Reply:
The concern is to not pay for any of this stuff, even in debug modes, for 99.999999% of atomics which don't have any padding bits. (as in, I have literally never ever seen a production use of an atomic with padding bits) The important ones are all `atomic<arithmetic type>` or `atomic<pointer>` and those shouldn't pay any cost here.

Reply:
Ok, if `std::has_unique_object_representations` is back, and us used with `constexpr`, and floating is checked as well, then no normal type is penalized at all in debug mode. Will it work for you?

Reply:
```
#if _HAS_CXX20
template <class _Ty>
bool _Cmpxchg_has_padding_bits_v = !has_unique_object_representations_v<_Ty> && !is_floating_point_v<_Ty>;
#endif
```
That's probably it, both type traits helpers are already visible in `<atomic>`

Reply:
Applied the change, please see if it is fine now